### ISSUE TITLE
Parallax Backgrounds weird behavior.

### ISSUE THREAD

**Issue by strellydev** [godotengine/godot#75562]
### Godot version

Godot 4.0.1

### System information

Windows 10

### Issue description

Hello everyone I hope you are having a nice day.

I'm porting a game from Godot 3 to Godot 4. In it, there are many parallax backgrounds with layers. However, in Godot 4, the position of the layers is messed up.

I don't know if this is a change in behavior of the parallax backgrounds that was planned for Godot 4, or if it's a bug, I tried asking in the Godot Q&A site and users answered telling me that there shouldn't be a difference in behavior, so I decided to post this as a bug. If it isn't please let me know.

I will upload two minimal reproduction projects, one for Godot 3, and one for Godot 4, so you can see the difference between versions. Both setups are exactly the same and you can move the camera with "WASD".

I'll also leave two pictures so you can have an idea of what I'm talking about.

Godot 3:
![GoodParallax](https://user-images.githubusercontent.com/113683950/229290807-51263872-ef97-4e50-a52b-317850fa4791.PNG)

Godot 4:
![BadParallax](https://user-images.githubusercontent.com/113683950/229290823-736f9448-13f9-4486-a6a4-e81c44388b35.PNG)

### Steps to reproduce

Create a parallax background and add layers to it.

### Minimal reproduction project

[ParallaxIssueGodot3.zip](https://github.com/godotengine/godot/files/11129580/ParallaxIssueGodot3.zip)
[ParallaxIssueGodot4.zip](https://github.com/godotengine/godot/files/11129581/ParallaxIssueGodot4.zip)

**Comment by strellydev** [comment]
Hello again, I was going around the source code of the engine, specifically in the ParallaxBackground.cpp and ParallaxLayer.cpp files, comparing them to the Godot 3 source code, and I noticed that in ParallaxLayer.cpp, function 
`void ParallaxLayer::set_base_offset_and_scale(const Point2 &p_offset, real_t p_scale)`
the screen_offset isn't being passed anymore and it isn't being taken into account for the position formula of the parallax layer. I tried re-adding the screen_offset parameter, and passing it from the ParallaxBackground.cpp call of the function
`void ParallaxBackground::_update_scroll()`
And it the parallax backgrounds started working fine again. So that seems to be the problem, I'll copy paste here the full new function so you get what I mean:
```
void ParallaxLayer::set_base_offset_and_scale(const Point2 &p_offset, real_t p_scale, const Point2 &p_screen_offset) {
	if (!is_inside_tree()) {
		return;
	}
	if (Engine::get_singleton()->is_editor_hint()) {
		return;
	}

	Point2 new_ofs = (p_screen_offset + (p_offset - p_screen_offset) * motion_scale) + motion_offset * p_scale + orig_offset * p_scale;

	if (mirroring.x) {
		real_t den = mirroring.x * p_scale;
		new_ofs.x -= den * ceil(new_ofs.x / den);
	}

	if (mirroring.y) {
		real_t den = mirroring.y * p_scale;
		new_ofs.y -= den * ceil(new_ofs.y / den);
	}

	set_position(new_ofs);
	set_scale(Vector2(1, 1) * p_scale * orig_scale);

	_update_mirroring();
}
```
The Point2 new_ofs formula is what I changed, and the new parameter is the p_screen_offset, which in the function declaration I made it so it has a default value of (0,0) in case there were other calls I didn't know about to the function.

So, why was this changed exactly? Could I leave it as it was in Godot 3 and make a pull request? or was there a reason for this?

Thanks a lot.
